### DISCUSSION TITLE
ESP32 WebDAV Tutorial on Fysetc SD-Wifi-Pro

### DISCUSSION THREAD

**Discussion by y2kdread** [RealTimeLogic/Xedge32#17] [Q&A]
I've been trying to follow the guide to setup WebDAV on ESP32 and hit a snag where I get the following error:

E (1547) cpu_start: External RAM could not be added to heap!

I am trying to flash it on the following ESP32 device:

https://www.fysetc.com/products/fysetc-upgrade-sd-wifi-pro-with-card-reader-module-run-wireless-by-esp32-chip-web-server-reader-uploader-3d-printer-parts)

I want to use this SD card and WebDAV to sync up CPAP data, but that SD card does not support WebDAV right now. Is it possible to modify the code so that it can map the storage that is on the SD card?

**Comment by surfskidude** [comment]
Your device does not come with the required 4 MB PSRAM. See details here: https://realtimelogic.com/ba/ESP32/source/GettingStarted.html

Consider using an ESP32-S3 with Ethernet if you plan on using WebDAV, e.g. https://www.aliexpress.us/item/3256805835507288.html

**Reply by y2kdread** [reply]
On the information provided by Fysetc, it mentions that it does have 2MB SPI PSRAM on the card. Is there any way I could modify the code to allow it to work with that lower PSRAM?

I am not much of a developer and haven't really worked with the ESP32, so my apologies for these basic questions.

**Reply by surfskidude** [reply]
No
